command/server.go

```diff
@@ -971,7 +984,7 @@ func configureDevTLS(c *ServerCommand) (func(), *server.Config, string, error) {
 				return nil, nil, certDir, err
 			}
 		}
-		config, err = server.DevTLSConfig(devStorageType, certDir)
+		config, err = server.DevTLSConfig(devStorageType, certDir, c.flagDevTLSSans)
```

Comment:
One comment if you've got a minute: do you want to add the listen address here as well? I think it'd help close https://github.com/hashicorp/vault/issues/18259, though we could always require the explicit SAN if we preferred... Just thinking the UX might be nice of the address+dev-tls (without dev-tls-san), but my 2c. :-)

Reply:
Nice idea! Added in 3deacb8c8e4323954b2c0e16613aea9d401f730a